lib/bloc/sentiment/sentiment_bloc.dart

```diff
@@ -22,11 +24,21 @@ class SentimentBloc extends Bloc<SentimentEvent, SentimentState> {
     try {
       emit(LoadingSentimentState());
       final positiveSentiment = await _sentimentRepo.getSentiment(event.query);
+      final negativeSentiment = positiveSentiment;
+      String rawCompletion = await openAIService.submitPrompt(event.query);
+      final sentimentOverview = rawCompletion.trim();
       if (positiveSentiment != null) {
         final negativeSentiment = 100 - positiveSentiment;
         emit(LoadedSentimentState(
           positiveSentiment: positiveSentiment,
           negativeSentiment: negativeSentiment,
+          sentimentOverview: sentimentOverview,
+        ));
+      } else if (positiveSentiment == null && negativeSentiment == null) {
```

Comment:
We should emit an `FailedSentimentState` whenever `positiveSentiment` is null because it means nothing was returned by the API, so it does not make sense to emit a loaded state. Please remove line 37 to 42.

Reply:
In the case where  `await _sentimentRepo.getSentiment(event.query);` is succeeding but `await openAIService.submitPrompt(event.query);` is failing, it would still throw an error despite the other succeeding , which is unfortunate.
 So I think it's better we seperate the sentiment overview functionality into its own bloc called `SentimentOverviewBloc` and leave `SentimentBloc` untouched but carry over my previous feedback to the new bloc.  You can take reference from `SentimentBloc` as its very similar